tests/test_cli.py

```diff
@@ -278,6 +292,18 @@ def test_eol__check_crlf(tmp_path):
     assert run((str(file_path), "--check", "--end-of-line=crlf")) == 0
 
 
+def test_eol__check_keep_lf(tmp_path):
+    file_path = tmp_path / "test.md"
+    file_path.write_bytes(b"lol\n")
+    assert run((str(file_path), "--check", "--end-of-line=keep")) == 0
```

Comment:
I think there's work to be done in the `if opts["check"]:` block.

Try adding this test. I suspect it will fail right now until the aforementioned is fixed
```suggestion

    file_path.write_bytes(b"lol\n")
    assert run((str(file_path), "--check", "--end-of-line=keep")) == 0
    
    file_path.write_bytes(b"mixed\nEOLs\r")
    assert run((str(file_path), "--check", "--end-of-line=keep")) == 1
```

Reply:
I've committed this and the fix to `if opts["check"]:`.

I'll release this today.